### PR TITLE
Merge IProvider and IApiProvider into INetworkProvider - part 1

### PR DESCRIPTION
**Pull request by andreibancioiu**
Merge `IProvider` and `IApiProvider` into `INetworkProvider` - part 1 (there will be another PRs on this topic).

Non breaking change (yet).

### CODE REVIEW

**Review by claudiu725 — DISMISSED** [review]

**Review by danielailie — APPROVED** [review]

File: src/networkProvider/proxyNetworkProvider.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+            let url = `${this.url}/${resourceUrl}`;
+            let response = await axios.get(url, this.config);
+            let payload = response.data.data;
+            return payload;
+        } catch (error) {
+            this.handleApiError(error, resourceUrl);
+        }
+    }
+
+    private async doPost(resourceUrl: string, payload: any): Promise<any> {
```
Comment (ccorcoveanu) [inline src/networkProvider/proxyNetworkProvider.ts:165]: Should we have a TODO or something to also remove duplicate code between these providers?

File: src/account.ts [hunk 2]
```diff
@@ -115,8 +115,8 @@ export class AccountOnNetwork {
         result.address = new Address(payload["address"] || 0);
         result.nonce = new Nonce(payload["nonce"] || 0);
         result.balance = Balance.fromString(payload["balance"]);
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply src/account.ts:117]: Fixed.

File: src/interface.networkProvider.ts [hunk 3]
```diff
@@ -0,0 +1,229 @@
+import { BigNumber } from "bignumber.js";
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply src/interface.networkProvider.ts:1]: I wanted it to stay near the classes that will use the interfaces (not in this PR, but in the next ones, that will remove the old network providers), not near the classes that implement those interfaces. But indeed, for the moment being, moved the file in `networkProvider`.

File: src/networkProvider/proxyNetworkProvider.ts (hunk 1 above)
Comment (andreibancioiu) [reply src/networkProvider/proxyNetworkProvider.ts:165]: While there is symmetry between the two providers, most functions are different (due to response enveloping, types of fields, URLs etc.). However, added TODOs (there are some duplicated chunks).

**Review by ccorcoveanu — APPROVED** [review]

**Review by danielailie — APPROVED** [review]

**Pull request merged**